tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteConsumerGroupsTest.java

```diff
@@ -17,279 +17,375 @@
-            "The consumer group deletion did not work as expected");
+    @ClusterTemplate("generator")
+    public void testDeleteWithUnrecognizedNewConsumerOption() {
+        String[] cgcArgs = new String[]{"--new-consumer", "--bootstrap-server", cluster.bootstrapServers(), "--delete", "--group", getDummyGroupId()};
+        assertThrows(OptionException.class, () -> getConsumerGroupService(cgcArgs));
     }
 
-    @ParameterizedTest
-    @ValueSource(strings = {"zk", "kraft"})
-    public void testDeleteWithMixOfSuccessAndError(String quorum) throws Exception {
-        createOffsetsTopic(listenerName(), new Properties());
-        String missingGroup = "missing.group";
+    private String getDummyGroupId() {
+        return composeGroupId(null);
+    }
 
-        // run one consumer in the group
-        ConsumerGroupExecutor executor = addConsumerGroupExecutor(1);
-        String[] cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--delete", "--group", GROUP};
-        ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
+    private String composeGroupId(GroupProtocol protocol) {
+        String groupPrefix = "test.";
+        return protocol != null ? groupPrefix + protocol.name : groupPrefix + "dummy";
+    }
 
-        TestUtils.waitForCondition(
-            () -> service.listConsumerGroups().contains(GROUP) && Objects.equals(service.collectGroupState(GROUP).state, "Stable"),
-            "The group did not initialize as expected.");
+    private String composeTopicName(GroupProtocol protocol) {
+        String topicPrefix = "foo.";
+        return protocol != null ? topicPrefix + protocol.name : topicPrefix + "dummy";
+    }
 
-        executor.shutdown();
+    private String composeMissingGroupId(GroupProtocol protocol) {
+        String missingGroupPrefix = "missing.";
+        return protocol != null ? missingGroupPrefix + protocol.name : missingGroupPrefix + "dummy";
+    }
 
-        TestUtils.waitForCondition(
-            () -> Objects.equals(service.collectGroupState(GROUP).state, "Empty"),
-            "The group did not become empty as expected.");
+    private AutoCloseable consumerGroupClosable(GroupProtocol protocol, String groupId, String topicName) {
+        boolean isNewCoordinator = cluster.config()
+                .serverProperties()
+                .get(NEW_GROUP_COORDINATOR_ENABLE_CONFIG)
+                .equals("true");
+
+        if (isNewCoordinator) {
+            return ConsumerGroupExecutor.buildConsumerGroup(
+                    cluster.bootstrapServers(),
+                    1,
+                    groupId,
+                    topicName,
+                    protocol.name,
+                    Optional.empty(),
+                    emptyMap(),
+                    false
+            );
+        }
+
+        return ConsumerGroupExecutor.buildClassicGroup(
```

Comment:
All cases we want to test are shown below.

1. zk + old consumer
2. kraft + old group coordinator + old consumer
3. kraft + new group coordinator + old consumer
4. kraft + new group coordinator + new consumer
5. combined kraft + old group coordinator + old consumer
6. combined kraft + new group coordinator + old consumer
7. combined kraft + new group coordinator + new consumer

In this path, we will run case2 and case5 two times since `groupProtocols` is consist of `CLASSIC` and `CONSUMER`. Hence, we should create `groupProtocols` according to `isNewCoordinator`. For example:

```java
        this.groupProtocols = cluster.config().serverProperties()
                .get(NEW_GROUP_COORDINATOR_ENABLE_CONFIG).equals("true")
                ? Arrays.asList(CLASSIC, CONSUMER)
                : singletonList(CLASSIC);
```